### PR TITLE
A new dawn

### LINKED ISSUE #131: Test service environment variables
For example, take a Twitter oauth credentials. The microservice would design a simple test to make sure the variables are valid. This test would be included in the `microservice.yml`

```yaml
tests:
  environment: 'python tests/environment.py'
```

The `tests/environment.py` would simple ping Twitter and exit 1 if the variables are not setup properly.

### LINKED ISSUE #220: omg validate incorrect output handling
When the `omg validate` gets output it adds numbers in a strange way.

 ```
1. Start of a really long sentence that 
2. continues but for some reason breaks into 2 lines
```

> Trying to replicate how I got here, but it's real 😉

### LINKED ISSUE #226: regression: building docker image startup takes > 5s
- it's no longer streaming the logs from docker
- it takes multiple seconds even for a cached rebuild

### LINKED ISSUE #249: Stream docker output to the CLI client (and UI)
tl;dr: we shouldn't wait until the entire command is finished, but output new lines whenever they are written to the stream.

### LINKED ISSUE #260: getOpenPort might return a blocked port
https://github.com/microservices/omg/blob/4b0d646d0c442e9595df814dfe438917e2fc32da/packages/omg/src/commands/Subscribe.ts#L54-L55

tl;dr: the returned port might have been taken by another program in the mean time. This should be handled with at least a graceful few retries.
I know this small deadlock isn't a huge problem, but the more OMG gets used, the higher the chances of users running into this.

### LINKED ISSUE #264: add help property to forwards
In order to match the [documentation](https://deploy-preview-78--microservice-guide.netlify.com/schema/forwards/#http), we need to add the support for help (with is optionnal) in the [schema](https://github.com/microservices/omg/blob/ec11b21b0757aa3f2e9f7851878fd2257211336c/packages/omg-validate/src/schema/schemas/forward.ts#L3)

We also need to add this help field inside the `forwards` view in the OMG APP

### LINKED ISSUE #265: Add support for arguments in the HTTP headers
https://github.com/microservices/microservice.guide/issues/105

### LINKED ISSUE #266: Add support for absolute URLs
https://github.com/microservices/microservice.guide/issues/104

### LINKED ISSUE #269: Refactoring
Internal issue to keep track of things

### PR DESCRIPTION
**Pull request by StoryscriptHuman**
Rewrite of the App/Validator/UI
Provides better stability and maintainability

WIP:
 - [x] Implement new UI design

Fixes #270 
Fixes #269
Fixes #266
Fixes #265
Fixes #264
Fixes #260
Fixes #249
Fixes #226
Fixes #220
Fixes #131

How to test:

```
# Git clone this repository then:
$ yarn
$ yarn build
$ cd packages/omg; yarn link

# Then in the project you want to test:
$ omg list
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -5,8 +5,8 @@ node_js:
 services:
   - docker
 
-before_install:
-  - npm install yarn; yarn link-bin && yarn install && yarn build
+before_script:
+  - yarn build && (cd packages/omg; yarn link)
```
Comment (matthewhudson) [inline .travis.yml]: Travis is failing here.
